membership-attribute-service/app/controllers/AttributeController.scala

```diff
@@ -68,10 +68,12 @@ class AttributeController {
       sfUser <- OptionT(tp.contactRepo.get(user))
       subscription <- OptionT(tp.subService.findByProductFamily(sfUser, productFamily))
       stripeCardToken <- OptionT(Future.successful(updateForm.bindFromRequest().value))
-      _ <- OptionT(tp.subService.setPaymentCardWithStripeToken(subscription.accountId, stripeCardToken))
+      updateResult <- OptionT(tp.subService.setPaymentCardWithStripeToken(subscription.accountId, stripeCardToken))
       pmNow <- OptionT(tp.subService.getPaymentCardByAccount(subscription.accountId))
-    } yield Ok(Json.obj("last4" -> pmNow.lastFourDigits, "cardType" -> pmNow.cardType, "type" -> pmNow.cardType)))
-      .run.map(_.getOrElse(notFound))
+    } yield updateResult match {
+      case CardUpdateSuccess => Ok(Json.obj("last4" -> pmNow.lastFourDigits, "cardType" -> pmNow.cardType, "type" -> pmNow.cardType))
+      case CardUpdateFailure(errType, errMsg, errCode) => Forbidden(Json.obj("type" -> errType, "message" -> errMsg, "code" -> errCode))
```

Comment:
At a later date, it's worth checking that these are all the fields that Stripe passes back. I know that for some requests it gives a `param` field which indicates which card param caused the error (e.g. `cvc`, `exp_month`), which can be useful to the client.